test/functional/eval/string_spec.lua

```diff
@@ -0,0 +1,175 @@
+local helpers = require('test.functional.helpers')
+local clear = helpers.clear
+local eq = helpers.eq
+local command = helpers.command
+local meths = helpers.meths
+local eval = helpers.eval
+local exc_exec = helpers.exc_exec
+local redir_exec = helpers.redir_exec
+local funcs = helpers.funcs
+local write_file = helpers.write_file
+
+describe('string() function', function()
+  before_each(clear)
+
+  describe('used to represent floating-point values', function()
+    it('dumps NaN values', function()
+      eq('str2float(\'nan\')', eval('string(str2float(\'nan\'))'))
+    end)
+
+    it('dumps infinite values', function()
+      eq('str2float(\'inf\')', eval('string(str2float(\'inf\'))'))
+      eq('-str2float(\'inf\')', eval('string(str2float(\'-inf\'))'))
+    end)
+
+    it('dumps regular values', function()
+      eq('1.5', funcs.string(1.5))
+      eq('1.56e-20', funcs.string(1.56000e-020))
+      eq('0.0', eval('string(0.0)'))
+    end)
+
+    it('dumps values with at most six digits after the decimal point',
+    function()
+      eq('1.234568e-20', funcs.string(1.23456789123456789123456789e-020))
+      eq('1.234568', funcs.string(1.23456789123456789123456789))
+    end)
+
+    it('dumps values with at most seven digits before the decimal point',
+    function()
+      eq('1234567.891235', funcs.string(1234567.89123456789123456789))
+      eq('1.234568e7', funcs.string(12345678.9123456789123456789))
+    end)
+
+    it('dumps negative values', function()
+      eq('-1.5', funcs.string(-1.5))
+      eq('-1.56e-20', funcs.string(-1.56000e-020))
+      eq('-1.234568e-20', funcs.string(-1.23456789123456789123456789e-020))
+      eq('-1.234568', funcs.string(-1.23456789123456789123456789))
+      eq('-1234567.891235', funcs.string(-1234567.89123456789123456789))
+      eq('-1.234568e7', funcs.string(-12345678.9123456789123456789))
```

Comment:
Does using lua numbers instead of `eval('string(-1...)')` open up this test to platform variability?